### PR TITLE
feat(@vtmn/svelte): add `VtmnDropdown` component

### PR DESCRIPTION
**Pull request by jb-thery**
## Changes description

This P.R adds the Svelte dropdown component

## Context

Related issue [#364](https://github.com/Decathlon/vitamin-web/issues/364)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on related showcases.
- [x] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- No

## Other information

Screenshot 2022-03-17 at 20 08 46

### CODE REVIEW

**Comment by jb-thery** [comment]
@gaspard-mathon to finish this component I need css classes that would allow me to modify the starting position of the menu (top, right; bottom, left), this way if the menu comes out of the viewport we can make it open in the opposite direction

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (lauthieb) [inline packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:103]: Don't you think it could be better to use a prefix before event names like "clickOutside" or not?

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (Tlahey) [inline packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:12]: Good PR!
But I'm not fan to use options. We are limited to the style applied on the element.
For example, if we need to have a skeleton on the content, it is not possible.

Same here, if we want to write something on right
image

For me, it's better to have a `VtmnDropdownItems` and do stuff like that : 

```javascript
<VtmnDropdown>
    <VtmnDropdownItem value=""></VtmnDropdownItem>
    <VtmnDropdownItem value="1">Option 1</VtmnDropdownItem>
    <VtmnDropdownItem value="2">Option 2</VtmnDropdownItem>
</VtmnDropdown>
```

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (Tlahey) [inline packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:116]: Please don't apply style directly on the div, use a class instead and apply a --var like Skeleton -> https://github.com/Decathlon/vitamin-web/pull/1047/files

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (jb-thery) [reply packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:103]: All right I update this

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (Tlahey) [inline packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:123]: Use `VtmnDivider` component

File: packages/sources/svelte/src/utils/clickOutside.js
Comment (Tlahey) [inline packages/sources/svelte/src/utils/clickOutside.js:1]: move this file into a `directive` folder

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (Tlahey) [inline packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:98]: What happening when you open a dropdown displayed on the bottom of the page. If the bottom are greater than the `document.body.scrollHeight` ?

image

@lauthieb is the use case is correct ? 
- When the dropdown is greater than the pageHeight, it should be displayed above the input
- When the dropdown is grater than the innerHeight but under the pageHeight, it should be displayed under the input (like dkt website)

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (Tlahey) [inline packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:13]: https://www.decathlon.design/726f8c765/p/42d824-dropdown/b/6647a9

I think, the dropdown component should avec an 'error' export 
image

Comment (jb-thery) [reply packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:13]: Dropdown error css class is depreciated from vitamin css

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (lauthieb) [reply packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:103]: @N0t-Simon what's your opinion please?

**Comment by lauthieb** [comment]
Thanks @jb-thery, for the moment I'm waiting for designers' answer. But after, we will be able to merge your work soon! :)

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (jb-thery) [reply packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:64]: @lauthieb all right, I await yout reply for finish the component

Comment (N0t-Simon) [reply packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:64]: hey @jb-thery, what you think about that? 

https://user-images.githubusercontent.com/50582328/159956259-418a27ff-c63f-47c8-a2c4-7a296310392c.mov

Comment (jb-thery) [reply packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:64]: Yes it's a nice option

Comment (lauthieb) [reply packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:64]: @jb-thery for the moment I convert to draft. Feel free to make this PR "ready for review" again when you will have made the implementation proposed by @N0t-Simon.

File: packages/showcases/core/csf/components/actions/dropdownItem.csf.js
Comment (lauthieb) [inline packages/showcases/core/csf/components/actions/dropdownItem.csf.js:47]: Is it possible to avoid this? We didn't create a csf for list-item for example.
Also I don't really understand the purpose of having a specific story for `DropdownItem`? I understood for `ListItem` but here I don't really find the value. 
Thanks in advance @jb-thery!

**Review by lauthieb — CHANGES_REQUESTED** [review]

File: packages/showcases/core/csf/components/actions/dropdown.csf.js
Comment (lauthieb) [inline packages/showcases/core/csf/components/actions/dropdown.csf.js:12]: can you please use `labelText` instead as we did with select, text input, checkbox, radio button and toggle? Thanks.

**Review by lauthieb — CHANGES_REQUESTED** [review]

File: packages/showcases/core/csf/components/actions/dropdownItem.csf.js
Comment (lauthieb) [inline packages/showcases/core/csf/components/actions/dropdownItem.csf.js:12]: I think `text` should be better here for a single dropdown item.

File: packages/sources/svelte/rollup.config.js
Comment (lauthieb) [inline packages/sources/svelte/rollup.config.js:14]:
```suggestion
        'VtmnDropdown',
        'VtmnDropdownItem',
```

**Review by lauthieb — CHANGES_REQUESTED** [review]

File: packages/showcases/core/csf/components/actions/dropdownItem.csf.js
Comment (Tlahey) [reply packages/showcases/core/csf/components/actions/dropdownItem.csf.js:12]: For me, I prefer use a slot for the label 🤔

File: packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte
Comment (Tlahey) [inline packages/sources/svelte/src/components/actions/VtmnDropdown/VtmnDropdown.svelte:72]: id = label text ?

**Comment by Tlahey** [comment]
I close this PR, rebase have issues, a new PR has been created

**Pull request closed without merging**
